iocore/hostdb/HostDB.cc

```diff
@@ -1175,12 +1174,13 @@ HostDBContinuation::dnsEvent(int event, HostEnt *e)
     timeout = nullptr;
   }
   EThread *thread = mutex->thread_holding;
-  if (event == EVENT_INTERVAL) {
+  if (event != DNS_EVENT_LOOKUP) {
+    // This was an event_interval or an event_immediate
+    // Either we timed out, or remove_trigger_pending gave up on us
     if (!action.continuation) {
       // give up on insert, it has been too long
-      // remove_trigger_pending_dns will notify and clean up all requests
-      // including this one.
-      remove_trigger_pending_dns();
+      hostDB.pending_dns_for_hash(hash.hash).remove(this);
+      hostdb_cont_free(this);
```

Comment:
hmm, should we hold the mutex before free this?

Reply:
If it was being called through the event processing logic, the mutex should already be held.  Let me take a look through on how it is being called directly.

Reply:
Looks like when we call handleEvent we may check for affinity threads but we don't grab the continuation lock. May need to do that.